src/main/java/org/hibernate/models/internal/IsResolvedTypeSwitch.java

```diff
@@ -63,7 +67,7 @@ public Boolean caseParameterizedType(
 			SourceModelBuildingContext buildingContext) {
 		final List<TypeDetails> typeArgs = parameterizedType.getArguments();
 		for ( TypeDetails arg : typeArgs ) {
-			if ( !isBound( arg, buildingContext ) ) {
+			if ( !isResolved( arg, buildingContext ) ) {
```

Comment:
Since `IsResolvedTypeSwitch` must never use the `SourceModelBuildingContext` in its implementation, you can simplify this to:
```suggestion
			if ( !arg.isResolved() ) {
```

Reply:
Thanks @mbladel !